src/main/src/main/java/org/geoserver/security/urlchecks/AbstractURLCheck.java

```diff
@@ -43,20 +43,24 @@ public void setDescription(String description) {
         this.description = description;
     }
 
-    /** @return the enable */
+    /** @return the enabled status */
     @Override
-    public boolean isEnabled() {
+    public Boolean isEnabled() {
         return enabled;
     }
 
-    /** @param enabled the enable to set */
-    public void setEnabled(boolean enabled) {
+    /** @param enabled the enabled status to set */
+    public void setEnabled(Boolean enabled) {
         this.enabled = enabled;
     }
 
+    /** @return the configuration of this checker */
+    public abstract String getConfiguration();
+
     /**
-     * A short string representation of the configuration of this checker, to be used for generic
-     * display purposes
+     * Sets the configuration of this checker.
+     *
+     * @param configuration the configuration to set
      */
-    public abstract String getConfiguration();
+    public abstract void setConfiguration(String configuration);
```

Comment:
The existing API has been misunderstood and broken here.
The existing configuration javadoc stated, configuration is a  "A short string representation of the configuration of this checker, to be used for generic display purposes". 
While the existing URL check happens to use a single configuration field that's a regex, future ones might have more complex configurations, and getConfiguration would still report a single, simple string.

Please make the REST API support the same opennes.